docs/src/userguide/language_basics.rst

```diff
@@ -366,14 +366,22 @@ of :ref:`error_return_values`.
 Error return values
 -------------------
 
-If you don't do anything special, a function declared with :keyword:`cdef` that
-does not return a Python object has no way of reporting Python exceptions to
-its caller. If an exception is detected in such a function, a warning message
-is printed and the exception is ignored.
+If you don't change the default behaviour, a function declared with :keyword:`cdef` or 
+:keyword:`cpdef` that does not return a Python object can not report its
+exception to the caller in an ordinary way, so a default value of ``0``/``False``
+will be immediately returned. If the caller is a Cython function, it will be
+detected as an exception and can be handled appropriately by the calling
+Cython function, but if the caller is a Python function it will only see a
+return value of ``0``/``False``.
```

Comment:
> If the caller is a Cython function, it will be
detected as an exception and can be handled appropriately by the calling
Cython function, but if the caller is a Python function it will only see a
return value of ``0``/``False``.

This isn't true. With no exception type set neither Cython nor Python callers can detect that an exception was set. (The "Exception ignored in..." message is generated within the function that raised the exception.)

Reply:
you are right